src/main/java/luffy/Storage.java

```diff
@@ -0,0 +1,102 @@
+package luffy;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+
+/**
+ * Storage class to handle save file.
+ *
+ * @author Silas Tay (A0233425M)
+ */
+public class Storage {
+    private String filePath;
+    private Ui ui;
+
+    /**
+     * Constructor for Storage class.
+     *
+     * @param filePath File path of save file
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        this.ui = ui;
+    }
+
+    /**
+     * Loads data from save file if exists.
+     *
+     * @return ArrayList of Tasks from save file
+     * @throws DukeException When there is an error loading the file
+     */
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> tasks = new ArrayList<Task>();
+
+        File dataFolder = new File("./data");
+        if (!dataFolder.exists()) {
+            dataFolder.mkdir();
+        }
+
+        File dataFile = new File(this.filePath);
+        try {
+            if (dataFile.exists()) {
+                Scanner scanner = new Scanner(dataFile);
+                while (scanner.hasNext()) {
+                    Task currentTask;
+                    String[] taskLine = scanner.nextLine().split(Pattern.quote(" | "));
+                    String taskDescription = taskLine[2];
+                    switch (taskLine[0]) {
+                        case "[T]":
```

Comment:
Just some small points on styling.
```suggestion
                    case "[T]":
```